index.js

```diff
@@ -1,42 +1,60 @@
 // ***** Question 1 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
-
-// console.log(counter) 
+console.log("%cQuestion 1", "color: red") 
+let counter = 1;
+counter = 2; //re-assign values
+console.log(counter) 
 // // => 2
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
-
+console.log("%cQuestion 2", "color: red") 
+const name = "Chris" //when using const, cant change/overwrite name. use const when you know for a fact 
+//the variable wont be changed
+console.log(name)//name is depracated. In the past in js library name was prob used. can be ignored
+// => "Chris"
 // console.log(name)
-// => "Raffy"
-// name = "Not Raffy" 
+// name = "Not Chris" 
 // => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red") 
 
-// console.log(drinkWater(12))
+function drinkWater(currentThirstLevel){ //use camelCase
+  console.log("Man I sure am thirsty") //console.log returns undefined. outputs to console in browser, puts returns in terminal
+  currentThirstLevel -= 1
+  console.log("Ahh that hits the spot")
+  return currentThirstLevel //js explicitly needs 'return' to return value in console
+}
+console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
-
+console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum){
```

Comment:
Nice! The key here was understanding the difference between `===` and `==`. Like you mention in the comments, you could also convert the string to a number with parseInt to avoid == :
Nice! Another solution would be to convert the string to a number with parseInt to avoid `==` :
`} else if (num === parseInt(maybeNum)) {`